src/inmanta/data/model.py

```diff
@@ -471,10 +471,27 @@ class PagingBoundaries:
     Represents the lower and upper bounds that should be used for the next and previous pages
     when listing domain entities.
 
-    :param start: largest value of the page for the primary sort column.
-    :param end: smallest value of the page for the primary sort column.
-    :param first_id: largest value of the page for the secondary sort column, if there is one.
-    :param last_id: smallest value of the page for the secondary sort column, if there is one.
+    The largest / smallest value of the current page represents respectively the min / max boundary value (exclusive) for the
+    neighbouring pages. Which represents next and which prev depends on sorting order (ASC or DESC).
+
+    Boundary values represent max and min values, regardless of sorting direction (ASC or DESC), i.e.
+                  |        prev             |                current page                    |    next
+     -------------|-------------------------|------------------------------------------------|----------------------
+     ASC sorting  | [  ...  ] (99 c)        | [ (100 d) ( 100 e) ... (10 000 r) (10 000 s) ] | (10 001 t) [ ...   ]
+                  |            start = 99   |     end = 100            start = 10 000        |     end = 10 001
+                  |            first_id = c |    last_id = d          first_id = s           |     first_id = t
+                  |
+     DESC sorting | [  ...  ] (10 001 s)    | [ (10 000 t) (10 000 r) ... (100 d) ( 100 c) ] |  (99 b) [ ...   ]
+                  |            end = 10 001 |          start = 10000          end = 100      |     start = 99
+                  |            last_id = t  |          first_id = t           last_id = c    |     first_id = b
```

Comment:
I'm not sure I understand what's conveyed here anymore. We should remove it if it adds more confusion than anything.
Alternatively we could maybe add a single scenario where we describe what happens in the simple case eg:

User requests all Resources with name  >= foo in ASC order. page size = 10
```
RequestPagingBoundary:
    start = foo
    end = None
```
    
Actual data = [foo1 ... foo10]

```
Pagingboundary:
    end = foo1
    start = foo10    # Reversed because these are meant to be used by the neighbouring pages
```
    
Does this make sense / is actually correct ?

Reply:
Yes, well explained, thank you!